src/crimson/osd/recovery_backend.cc

```diff
@@ -34,7 +34,7 @@ void RecoveryBackend::clean_up(ceph::os::Transaction& t,
   temp_contents.clear();
 
   for (auto& [soid, recovery_waiter] : recovering) {
-    if (recovery_waiter.obc) {
+    if (recovery_waiter.obc && recovery_waiter.obc->obs.exists) {
```

Comment:
what's the issue here? thanks

Reply:
obc may be created when osd trying to load it from disk while the object doesn't exist, under which circumstance the obc->obs.exists would be false. So a non-empty obc doesn't necessarily indicate an existing object

Reply:
Thanks (and I think this deserves a comment, with the text of your answer)

Reply:
it'd be better to extract this change into its own commit.